### PR TITLE
feat(pace-100): handles generic hierarchy

### PR DESCRIPTION
**Pull request by bvdeenen**

### CODE REVIEW

File: app/src/main/kotlin/com/getstrm/pace/domain/IntegrationClient.kt [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    open suspend fun platformResourceName(index: Int): String = "Level-$index"
+
+    open suspend fun createBlueprint(fqn: String): DataPolicy =
+        throwUnimplemented(
+            "createBlueprint from fully qualified name in integration $id of type ${this::class.simpleName}"
+        )
+
+    abstract suspend fun listDatabases(
```
Comment (bvdeenen) [inline app/src/main/kotlin/com/getstrm/pace/domain/IntegrationClient.kt:96]: do we still need these database, schema and table methods now that we have the legacy service.

File: app/src/main/kotlin/com/getstrm/pace/processing_platforms/bigquery/BigQueryClient.kt [hunk 2]
```diff
@@ -63,29 +69,39 @@ class BigQueryClient(
 
     private val bigQueryDatabase = BigQueryDatabase(this, config.projectId)
 
+    override suspend fun platformResourceName(index: Int): String {
+        return when (index) {
+            1 -> "project"
+            2 -> "dataset"
+            3 -> "table"
+            // TODO unpleasant client interaction
+            else -> "Level-$index"
```
Comment (bvdeenen) [inline app/src/main/kotlin/com/getstrm/pace/processing_platforms/bigquery/BigQueryClient.kt:79]: I think `super(index)` might be the best default handler.

File: app/src/main/kotlin/com/getstrm/pace/processing_platforms/databricks/DatabricksClient.kt [hunk 3]
```diff
@@ -48,6 +50,15 @@ class DatabricksClient(
             .setClientSecret(config.clientSecret)
             .let { config -> AccountClient(config) }
 
+    override suspend fun platformResourceName(index: Int): String {
+        return when (index) {
+            1 -> "catalog"
+            2 -> "schema"
+            3 -> "table"
+            else -> "Level-$index"
```
Comment (bvdeenen) [inline app/src/main/kotlin/com/getstrm/pace/processing_platforms/databricks/DatabricksClient.kt:58]: `super(index)`

File: app/src/main/kotlin/com/getstrm/pace/processing_platforms/snowflake/SnowflakeClient.kt [hunk 4]
```diff
@@ -42,6 +45,15 @@ class SnowflakeClient(override val config: SnowflakeConfig) : ProcessingPlatform
     // which is why we have a hardcoded one here
     val database = SnowflakeDatabase(this, config.database)
 
+    override suspend fun platformResourceName(index: Int): String {
+        return when (index) {
+            1 -> "warehouse"
+            2 -> "schema"
+            3 -> "table"
+            else -> "Level-$index"
```
Comment (bvdeenen) [inline app/src/main/kotlin/com/getstrm/pace/processing_platforms/snowflake/SnowflakeClient.kt:53]: `super(index)`

File: protos/getstrm/pace/api/processing_platforms/v1alpha/processing_platforms.proto [hunk 5]
```diff
@@ -114,7 +115,7 @@ message ListGroupsResponse {
 message GetBlueprintPolicyRequest {
     string platform_id = 1 [(buf.validate.field).string = {min_len: 1}];
     entities.v1alpha.Table table = 2;
-    string fqn = 3;
+    optional string fqn = 3;
```
Comment (ivan-p92) [inline protos/getstrm/pace/api/processing_platforms/v1alpha/processing_platforms.proto:118]: Should we perhaps name this one `integration_fqn`, as we do in `ResourceUrn`?

File: settings.gradle.kts [hunk 6]
```diff
@@ -17,6 +17,9 @@ pluginManagement {
         id("org.flywaydb.flyway") version "9.22.3"
     }
 }
+plugins {
+    id("org.gradle.toolchains.foojay-resolver-convention") version "0.5.0"
```
Comment (ivan-p92) [inline settings.gradle.kts]: Where was this needed for?

File: examples/standalone-many-tables/data-policy.yaml [hunk 7]
```diff
@@ -2,9 +2,6 @@ metadata:
   description: ""
   version: 1
   title: public.demo
-platform:
-  id: standalone-sample-connection
```
Comment (ivan-p92) [inline examples/standalone-many-tables/data-policy.yaml:6]: The platform has been removed, and is now part of `ResourceUrn`. However, I don't see any mention of the platform in these example policies anymore. How is that handled now?

File: protos/getstrm/pace/api/entities/v1alpha/entities.proto [hunk 8]
```diff
@@ -193,11 +205,7 @@ message DataPolicy {
   message Target {
     TargetType type = 1;
     // The full and unique name to be used in the target platform. E.g. the view name.
```
Comment (ivan-p92) [inline protos/getstrm/pace/api/entities/v1alpha/entities.proto:207]: A few of the comments where we now use `ResourceUrn` are outdated.

**Review by ivan-p92 — APPROVED** [review]

**Pull request merged**